cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -0,0 +1,244 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.api.schedule.SchedulableProgramType;
+import co.cask.cdap.api.schedule.ScheduleSpecification;
+import co.cask.cdap.app.ApplicationSpecification;
+import co.cask.cdap.app.runtime.ProgramController;
+import co.cask.cdap.app.runtime.ProgramRuntimeService;
+import co.cask.cdap.app.store.StoreFactory;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.namespace.NamespacedLocationFactory;
+import co.cask.cdap.config.PreferencesStore;
+import co.cask.cdap.data2.transaction.queue.QueueAdmin;
+import co.cask.cdap.gateway.auth.Authenticator;
+import co.cask.cdap.internal.app.runtime.schedule.ScheduledRuntime;
+import co.cask.cdap.internal.app.runtime.schedule.Scheduler;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramStatus;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.http.HttpResponder;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Lists;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.reflect.TypeToken;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.internal.RunIds;
+import org.jboss.netty.handler.codec.http.HttpHeaders;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Workflow HTTP Handler.
+ */
+@Singleton
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}")
+public class WorkflowHttpHandler extends ProgramLifecycleHttpHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(ProgramLifecycleHttpHandler.class);
+
+  @Inject
+  public WorkflowHttpHandler(Authenticator authenticator, StoreFactory storeFactory, WorkflowClient workflowClient,
+                             CConfiguration configuration, ProgramRuntimeService runtimeService,
+                             DiscoveryServiceClient discoveryServiceClient, QueueAdmin queueAdmin, Scheduler scheduler,
+                             PreferencesStore preferencesStore, NamespacedLocationFactory namespacedLocationFactory) {
+    super(authenticator, storeFactory, workflowClient, configuration, runtimeService, discoveryServiceClient,
+          queueAdmin, scheduler, preferencesStore, namespacedLocationFactory);
+  }
+
+  @POST
+  @Path("/apps/{app-id}/workflows/{workflow-name}/{run-id}/suspend")
```

Comment:
should send an API review for these APIs. Not RESTy, because of the missing "runs" component after ``{workflow-name}``.